animation_nodes/algorithms/mesh_generation/find_shortest_path.pyx

```diff
@@ -0,0 +1,112 @@
+from ... data_structures import GPStroke
+from ... data_structures cimport (
+    Mesh,
+    LongList,
+    ColorList,
+    FloatList,
+    PolySpline,
+    DoubleList,
+    BooleanList,
+    Vector3DList,
+)
+
+from libc.math cimport INFINITY
+from . line import getLinesMesh
+from ... math cimport distanceVec3
+
+# Dijkstra's algorithm (https://en.wikipedia.org/wiki/Dijkstra%27s_algorithm) but is implemented
+# such a way to handle multi-sources and mesh with multiple islands.
+def getShortestPath(Mesh mesh, LongList sources, LongList targets, str pathType, str mode):
+    cdef Vector3DList vertices = mesh.vertices
+    cdef long vertexCount = vertices.length
+    cdef float maxDistance = INFINITY
+    cdef DoubleList distances = DoubleList.fromValue(maxDistance, length = vertexCount)
+    cdef LongList previousVertices = LongList.fromValue(-1, length = vertexCount)
+    cdef BooleanList visitedVertices = BooleanList.fromValue(False, length = vertexCount)
+    cdef Py_ssize_t i, targetIndex
+
+    for i in range (sources.length):
+        distances.data[sources.data[i]] = 0.0
+
+    if mode == "PATH":
+        targetIndex = targets.data[0]
+    else:
+        targetIndex = <int>INFINITY
+
+    cdef float distance, minDistance
+    cdef Py_ssize_t k, l, currentIndex, neighboursStart, neighbourIndex
+    cdef LongList neighboursAmounts, neighboursStarts, neighbours
+    neighboursAmounts, neighboursStarts, neighbours = mesh.getLinkedVertices()[:3]
+
+    for i in range(vertexCount):
+
+        minDistance = maxDistance
+        for k in range(vertexCount):
+            distance = distances.data[k]
+            if distance < minDistance and not visitedVertices.data[k]:
+                minDistance = distance
+
+        if minDistance == maxDistance:
+            break
+
+        for currentIndex in range(vertexCount):
+            if abs(minDistance - distances.data[currentIndex]) < 1.0e-6:
+                visitedVertices.data[currentIndex] = True
+
+                neighboursStart = neighboursStarts.data[currentIndex]
+                for l in range(neighboursAmounts.data[currentIndex]):
```

Comment:
Don't use `l` as a variable name. `i, j, and k` are usually used.

Reply:
Done.